Description: Fix bash_completion invocation
Forwarded: not-needed
Author: Jérémy Lal <kapouer@melix.org>
Last-Update: 2011-04-30
--- a/lib/utils/completion.sh
+++ b/lib/utils/completion.sh
@@ -12,7 +12,7 @@
 export COMP_WORDBREAKS
 
 if complete &>/dev/null; then
-  _npm_completion () {
+  _npm () {
     local si="$IFS"
     IFS=$'\n' COMPREPLY=($(COMP_CWORD="$COMP_CWORD" \
                            COMP_LINE="$COMP_LINE" \
@@ -21,9 +21,9 @@
                            2>/dev/null)) || return $?
     IFS="$si"
   }
-  complete -F _npm_completion npm
+  complete -F _npm npm
 elif compctl &>/dev/null; then
-  _npm_completion () {
+  _npm () {
     local cword line point words si
     read -Ac words
     read -cn cword
@@ -38,6 +38,6 @@
                        2>/dev/null)) || return $?
     IFS="$si"
   }
-  compctl -K _npm_completion npm
+  compctl -K _npm npm
 fi
 ###-end-npm-completion-###
