Description: Change shell command from node to nodejs
Forwarded: not-needed
Author: Jérémy Lal <kapouer@melix.org>
Last-Update: 2012-09-21

--- a/test/packages/npm-test-shrinkwrap/test.sh
+++ b/test/packages/npm-test-shrinkwrap/test.sh
@@ -3,7 +3,7 @@
 # ensure that we get the npm being tested, not some global thing.
 npmbin=$npm_config_prefix/bin/npm
 npm () {
-  node $npmbin "$@"
+  nodejs $npmbin "$@"
 }
 
 # work around the weird env we're in, as part of npm's test
--- a/scripts/index-build.js
+++ b/scripts/index-build.js
@@ -1,4 +1,4 @@
-#!/usr/bin/env node
+#!/usr/bin/nodejs
 var fs = require("fs")
   , path = require("path")
   , cli = path.resolve(__dirname, "..", "doc", "cli")
--- a/scripts/clean-old.sh
+++ b/scripts/clean-old.sh
@@ -26,7 +26,7 @@
 else
   node="$NODE"
   if [ "x$node" = "x" ]; then
-    node=`which node`
+    node=`which nodejs`
   fi
   if [ "x$node" = "x" ]; then
     echo "Can't find node to determine prefix. Aborting."
--- a/scripts/install.sh
+++ b/scripts/install.sh
@@ -42,7 +42,7 @@
 export npm_config_loglevel
 
 # make sure that node exists
-node=`which node 2>&1`
+node=`which nodejs 2>&1`
 ret=$?
 if [ $ret -eq 0 ] && [ -x "$node" ]; then
   (exit 0)
--- a/Makefile
+++ b/Makefile
@@ -1,4 +1,5 @@
 SHELL = bash
+node = nodejs
 
 markdowns = $(shell find doc -name '*.md' | grep -v 'index') README.md
 
@@ -33,23 +34,23 @@
 	@echo "Installing latest published npm"
 	@echo "Use 'make install' or 'make link' to install the code"
 	@echo "in this folder that you're looking at right now."
-	node cli.js install -g -f npm
+	$(node) cli.js install -g -f npm
 
 install: all
-	node cli.js install -g -f
+	$(node) cli.js install -g -f
 
 # backwards compat
 dev: install
 
 link: uninstall
-	node cli.js link -f
+	$(node) cli.js link -f
 
 clean: doc-clean uninstall
 	rm npmrc
-	node cli.js cache clean
+	$(node) cli.js cache clean
 
 uninstall:
-	node cli.js rm npm -g -f
+	$(node) cli.js rm npm -g -f
 
 doc: $(mandocs) $(htmldocs)
 
@@ -89,17 +90,17 @@
 	scripts/doc-build.sh $< $@
 
 doc/cli/index.md: $(markdowns) scripts/index-build.js scripts/doc-build.sh package.json
-	node scripts/index-build.js > $@
+	$(node) scripts/index-build.js > $@
 
 node_modules/.bin/ronn:
-	node cli.js install https://github.com/isaacs/ronnjs/tarball/master
+	$(node) cli.js install https://github.com/isaacs/ronnjs/tarball/master
 
 doc: man
 
 man: $(cli_docs) $(api_docs)
 
 test:
-	node cli.js test
+	$(node) cli.js test
 
 version: link
 	git add package.json &&\
--- a/cli.js
+++ b/cli.js
@@ -1,2 +1,2 @@
-#!/usr/bin/env node
+#!/usr/bin/nodejs
 require("./bin/npm-cli.js")
--- a/bin/npm-cli.js
+++ b/bin/npm-cli.js
@@ -1,4 +1,4 @@
-#!/usr/bin/env node
+#!/usr/bin/nodejs
 ;(function () { // wrapper in case we're in module_context mode
 
 // windows: running "npm blah" in this folder will invoke WSH, not node.
--- a/lib/utils/exec.js
+++ b/lib/utils/exec.js
@@ -108,6 +108,9 @@
   if (!isNaN(gid)) opts.gid = gid
   if (!isNaN(opts.uid)) opts.uid = +opts.uid
   if (!isNaN(opts.gid)) opts.gid = +opts.gid
+  c = c.replace(/^\s*node(\s+|$)/gm, function(m, p1, off, s) {
+    return "nodejs" + (p1.length ? " " : "")
+  })
   var name = c +" "+ a.map(JSON.stringify).join(" ")
   log.silly([c, a, opts.cwd], "spawning")
   cp = child_process.spawn(c, a, opts)
